CRM/Civigiftaid/Utils/Contribution.php

```diff
@@ -95,19 +95,24 @@ public static function addContributionToBatch($contributionIDs, $batchID) {
     );
 
     foreach ($contributionIDs as $contributionID) {
-      //$batchContribution = new CRM_Core_DAO_EntityBatch( );
-      $batchContribution = new CRM_Batch_DAO_EntityBatch();
-      $batchContribution->entity_table = 'civicrm_contribution';
-      $batchContribution->entity_id = $contributionID;
-
       // check if the selected contribution id already in a batch
       // if not, add to batchContribution else keep the count of contributions that are not added
-
-      if ($batchContribution->find(TRUE)) {
+      $contributionNotInBatch = civicrm_api3('EntityBatch', 'get', [
+          'sequential' => 1,
+          'entity_table' => "civicrm_contribution",
+          'batch_id.type_id' => "giftaid_batch",
+          'entity_id' => $contributionID,
+      ]);
```

Comment:
Rather than make an API call per contribution ID. Perhaps we can have a function that uses a single API call to fetch the status of the contribution ID's first and we can use that in this function.

i.e 

```php
$contributionsNotInBatch = civicrm_api3('EntityBatch', 'get', [
          'return' => 'entity_id,
          'entity_table' => "civicrm_contribution",
          'batch_id.type_id' => "giftaid_batch",
          'entity_id' => [ 'IN' => [$contributionIDs]],
      ]);

```